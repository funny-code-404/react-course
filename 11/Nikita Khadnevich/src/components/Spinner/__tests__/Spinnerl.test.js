import React from 'react';
import { Provider } from 'react-redux';
import SpinnerHome from '../../Spinner/index';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });
import { connect } from 'react-redux';
import configureMockStore from 'redux-mock-store' //ES6 modules

const newTimeout = 6000;
jest.setTimeout(newTimeout) 

describe('<Spinner>', () => {
   const mockStore = configureMockStore()// create Stor
   const initialState = {
      civil: {isFetching: true},
      units: {isFetching: true},
      structures: {isFetching: true},
      technologies: {isFetching: true},
   }// Init StorContain
   const store = mockStore(initialState) // Put contain into Stor

   it('Is Spinner render correctly', () => {
      const props = { title: 'Render correctly!'}
      const wrapper = mount(
         <Provider store={store} >
            <SpinnerHome {...props}/>
         </Provider>
      )
   expect(wrapper).toMatchSnapshot()
   expect(wrapper).toHaveLength(1)
   })

   it('should get-Function called', () => {
      const fetchingState = jest.fn();
      const wrapper = mount(
         <Provider store={store} >
            <SpinnerHome fetchingState={fetchingState()}/>
         </Provider>
      )
      expect(fetchingState).toBeCalled()
   })

   it('should to find P-element is String', () => {
      const props = {
         title: 'Truthy'
      }
      const wrapper = mount(
         <Provider store={store} >
            <SpinnerHome {...props} />
         </Provider>
      )
      expect(typeof (wrapper.find("p").text()) == 'string').toBe(true)
   })

   it('check is P-element', () => {
      const props = {
            title: "Waiting"
      }
      const wrapper = mount(
         <Provider store={store} >
            <SpinnerHome {...props}/>
         </Provider>
      )
      expect(wrapper.find("p").text()).toEqual('Waiting');
   })
})