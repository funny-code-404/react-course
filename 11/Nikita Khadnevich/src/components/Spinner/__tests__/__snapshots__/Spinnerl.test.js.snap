// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner> Is Spinner render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SpinnerHome)
    title="Render correctly!"
  >
    <SpinnerHome
      dispatch={[Function]}
      isFetchingCivil={true}
      isFetchingStru={true}
      isFetchingTech={true}
      isFetchingUnit={true}
      title="Render correctly!"
    >
      <div
        className="blocks"
      >
        <p>
          Render correctly!
        </p>
        <div
          className="block orange"
        />
        <div
          className="block blue"
        />
      </div>
    </SpinnerHome>
  </Connect(SpinnerHome)>
</Provider>
`;
